build.gradle

```diff
@@ -80,20 +84,80 @@ project(":service") {
     shadowJar {
         archiveName = "commercetools-payone-integration.jar"
         manifest {
-            attributes 'Main-Class': 'com.commercetools.pspadapter.payone.ServiceFactory'
+            attributes 'Main-Class': project.mainClassName
+
+            // set version to define at runtime using getImplementationVersion()
+            attributes 'Implementation-Title': rootProject.name
+            attributes 'Implementation-Version': allprojects.version
+        }
+
+        doLast {
+            println "Packaged JAR to ${destinationDir}/${archiveName}"
+        }
+    }
+
+    run {
+        dependsOn ':service:convertGradlePropertiesToJvmArgs'
+    }
+
+    task convertGradlePropertiesToJvmArgs (type:Task) {
+        description 'Convert the gradle build properties (set in ~/.gradle/gradle.properties) ' +
+                    'to JVM runtime properties (-Dkey=value arguments)'
+
+        doLast {
+            applicationDefaultJvmArgs = assertAndGetServiceProperties(project)
         }
     }
 }
 
 /**
+ * Convert the list of required properties from gradle build properties or environment variables
+ * to java runtime properties (supplied as <b>-Dkey=value</b> arguments).
+ * <p>
+ * The function tries to find a property value from the {@code project} properties and then from environment variables.
+ * If neither of them exists - throw an exception (fail-fast)
+ * <p>
+ *     Opposite to {@link #assertExecutableSpecRequirementsForTests(org.gradle.api.Project, org.gradle.api.tasks.testing.Test)}
+ *     this function is used for the service runtime.
+ *
+ * @param project gradle project from which to take the properties
+ * @return a list of java runtime argument in <i>-Dkey=value</i> syntax.
+ *
+ * @throws GradleException if any required system property or environment variable is undefined or empty
+ * @see #assertExecutableSpecRequirementsForTests(org.gradle.api.Project, org.gradle.api.tasks.testing.Test)
+ */
+static List<String> assertAndGetServiceProperties(Project project) {
```

Comment:
Now the app requires 7 mandatory environment variables to run the app, but IDEs (at least Intellij IDEA, but guess the other too) have *very* unhandy and manual way to manage environment variables for an application: you set them one-by-one in grid window and there is no way to automate this process ([see this SO question](http://stackoverflow.com/questions/39766723/intellij-idea-run-debug-java-application-with-a-set-of-environment-variables)]

Because of this restriction the app development has the next workflow:
  1. Build the app
  2. Export environment variables
  3. Run the app from command line with special JVM debug arguments (`java -agentlib:jdwp=transport=dt_socket,server=y,suspend=n,address=1044 -jar service/build/libs/commercetools-payone-integration.jar`)
  4. Attach the debugger from IDE to the running JVM.

If you need to change one property - you edit file with exported environment variables and then repeat the steps 2-4 again.

Instead of this I'd like to make the project import-run-debug process easy to maintain (aka click-one-button), like it is set-up in the integration tests: copies values from `gradle.properties` settings to environment variables and runs the test. Unfortunately, this feature in gradle works only for test (or _exec_ task, see [this](http://stackoverflow.com/a/26634167/907576) and [this](http://stackoverflow.com/a/26634167/907576) ).

So, at last I found the gradle _application_ plugin which allows easily run a java app with some additional pre-configuration. That's why i copy the values from `gradle.properties` to java run arguments. 

Any way, i think it is _correct_ approach for normal java apps to allow override most of the properties by runtime arguments, isn't it?